packages/react/src/components/NumberInput/NumberInput.js

```diff
@@ -171,14 +156,16 @@ class NumberInput extends Component {
     translateWithId: id => defaultTranslations[id],
   };
 
-  /**
-   * The DOM node refernce to the `<input>`.
-   * @type {HTMLInputElement}
-   */
-  _inputRef = null;
-
   static getDerivedStateFromProps({ min, max, value = 0 }, state) {
     const { prevValue } = state;
+
+    if (typeof value === 'string' && value === '' && state.value !== '') {
```

Comment:
```suggestion
    if (!useControlledStateWithValue && value === '' && prevValue !== '') {
```

* Returning non-`null` value, which means changing the state to the returning value, whenever `value` prop is an empty string and `value` state is not an empty string causes canceling any state update (to non-empty string, e.g. by up/down buttons or user's type) if `value` prop is an empty string (note that `gDSFP` is called for user-initiated state updates, too). The problem is observed with `<NumberInput value="" />` where it doesn't allow user to make any edit at all. Wondering if you simply meant update in `value`prop to an empty string, which led to my suggestion above.
* `useControlledStateWithValue` flag must make the entire `gDSFP` no-op (by returning `null`). The purpose of flag `useControlledStateWithValue` is not doing any `value` state update from `value` prop
* `typeof value === 'string'` is redundant if we are doing strict equal check of `value` with an empty string (given strict equal check involves type checking)

Reply:
@abbeyhrt Any thoughts on this...? Thanks!

Reply:
I'll make the changes for the last two points, those seem good to me 👍 Could you help me understand your first point?  The intent behind it was to prevent anybody from setting the value to anything but an empty string or a number, could you provide steps for me to reproduce the problem if `value` is set to an empty string?